src/py42/services/storage/restore.py

```diff
@@ -0,0 +1,63 @@
+from py42.services import BaseService
+
+
+class RestoreService(BaseService):
+    def create_restore_session(
+        self,
+        device_guid,
+        data_key_token=None,
+        private_password=None,
+        encryption_key=None,
+    ):
+        """Creates a web restore connection.
+        See https://console.us.code42.com/apidocviewer/#WebRestoreSession
+        """
+        uri = u"/api/WebRestoreSession"
+        json_dict = {
+            u"computerGuid": device_guid,
+            u"dataKeyToken": data_key_token,
+            u"privatePassword": private_password,
+            u"encryptionKey": encryption_key,
+        }
+        return self._connection.post(uri, json=json_dict)
```

Comment:
I noticed while testing if an archive has a private_password and you pass the wrong one here, py42 raises a generic 500, but the error text is clear that the password is incorrect. Perhaps we could create a custom exception for these?

Reply:
Example exception w/ response text so you don't have to do the work to recreate it: 

```
>>> r
Py42InternalServerError('Failure in HTTP call 500 Server Error: Server Error for url: https://web-epf-sea.crashplan.com/api/WebRestoreSession')
>>> r.response.text
'[{"name":"PRIVATE_PASSWORD_INVALID","description":"An error has occurred. See server logs for more information.","objects":[]}]'
```

Reply:
And here's for if a custom encryption key doesn't match: 

```
>>> r.response.text
'[{"name":"CUSTOM_KEY_INVALID","description":"An error has occurred. See server logs for more information.","objects":[]}]'
```

Reply:
Done